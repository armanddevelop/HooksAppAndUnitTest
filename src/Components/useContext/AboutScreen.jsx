import { useContext } from "react";
import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { user, setUser } = useContext(UserContext);
  const userRest = {};
  return (
    <div>
      <h1>AboutScreen</h1>
      <hr />
      <pre>{JSON.stringify(user, null)}</pre>
      <button className="btn btn-warning" onClick={() => setUser(userRest)}>
        Log out
      </button>
    </div>
  );
};
