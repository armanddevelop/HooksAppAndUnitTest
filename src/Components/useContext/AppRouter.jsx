import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import { LognScreen, AboutScreen, HomeScreen, NavBar } from "./index";
export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <div className="container">
          <Routes>
            <Route exact path="/" element={<HomeScreen />} />
            <Route exact path="/about" element={<AboutScreen />} />
            <Route exact path="/login" element={<LognScreen />} />
            <Route path="*" element={<HomeScreen />} />
          </Routes>
        </div>
      </div>
    </Router>
  );
};
