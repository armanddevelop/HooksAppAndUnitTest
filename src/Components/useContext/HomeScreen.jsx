import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {
  const { user } = useContext(UserContext);
  //console.log("esto vale userContext ", user);
  return (
    <div>
      <h1>HomeScreen</h1>
      <hr />
      <pre>{JSON.stringify(user, null)}</pre>
    </div>
  );
};
