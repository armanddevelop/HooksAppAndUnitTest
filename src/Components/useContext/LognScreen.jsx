import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const LognScreen = () => {
  const { setUser } = useContext(UserContext);
  const userObj = {
    id: 123,
    name: "creting",
  };
  return (
    <div>
      <h1>Login Screen</h1>
      <hr />
      <button className="btn btn-primary" onClick={() => setUser(userObj)}>
        User
      </button>
    </div>
  );
};
