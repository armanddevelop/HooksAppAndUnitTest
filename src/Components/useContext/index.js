export { HomeScreen } from "./HomeScreen";
export { AboutScreen } from "./AboutScreen";
export { LognScreen } from "./LognScreen";
export { NavBar } from "./NavBar";
