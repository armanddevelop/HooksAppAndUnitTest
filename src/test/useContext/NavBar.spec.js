import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import toJson from "enzyme-to-json";
import { NavBar } from "../../Components/useContext/NavBar";

describe("Test to NavBar component", () => {
  const container = shallow(<NavBar />);
  it("snapshot to NavBar component", () => {
    expect(toJson(container)).toMatchSnapshot();
  });
  it("NavBar component contain NavLink", () => {
    const navLink = container.find("NavLink").exists();
    expect(navLink).toBeTruthy();
  });
});
