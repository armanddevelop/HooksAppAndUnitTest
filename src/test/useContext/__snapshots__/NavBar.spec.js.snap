// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to NavBar component snapshot to NavBar component 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <div
    className="container-fluid"
  >
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <NavLink
          aria-current="page"
          className="nav-link"
          exact="true"
          to="/"
        >
          Home
        </NavLink>
        <NavLink
          className="nav-link"
          exact="true"
          to="/login"
        >
          Login
        </NavLink>
        <NavLink
          className="nav-link"
          exact="true"
          to="/about"
        >
          About
        </NavLink>
      </div>
    </div>
  </div>
</nav>
`;
